src/generate_tables.nr

```diff
@@ -0,0 +1,238 @@
+                        let bit6out = ((bit6 & 3) > 1) as u32;
+
+                        // let mut index = bit1;
+                        // index *= 7;
+                        let mut index = bit2;
+                        index *= 4;
+                        index += bit3;
+                        index *= 4;
+                        index += bit4;
+                        index *= 4;
+                        index += bit5;
+                        index *= 4;
+                        index += bit6;
+
+                        // let mut value = bit1out;
+                        // value *= 2;
+                        let mut value = bit2out;
+                        value *= 2;
+                        value += bit3out;
+                        value *= 2;
+                        value += bit4out;
+                        value *= 2;
+                        value += bit5out;
+                        value *= 2;
+                        value += bit6out;
+
+                        result[index] = value as Field;
+                    }
+                }
+            }
+            // }
+        }
+    }
+    println(
+        f"pub global BASE4_MAJORITY_DECODE_5BIT_TABLE: [Field; 1024] = {result}",
+    );
+}
+
+#[test]
+fn print_xor_decoding_table() {
+    //Safety:
+    unsafe { __compute_xor_decoding_table() };
+}
+
+#[test]
+fn print_partial_choose_decoding_table() {
+    //Safety:
+    unsafe { __compute_partial_choose_decoding_table() };
+}
+
+#[test]
+fn print_majority_decoding_table() {
+    //Safety:
+    unsafe { __compute_majority_decoding_table() };
+}
+
+#[test]
+fn print_encoding_table() {
+    // Safety: no need to constrain this we're just computing+printing lookup table values
+    unsafe {
+        let t0 = compute_base4_conversion_table::<2>();
+        let t1 = compute_base4_conversion_table::<4>();
+        let t2 = compute_base4_conversion_table::<8>();
+        let t3 = compute_base4_conversion_table::<16>();
+        let t4 = compute_base4_conversion_table::<32>();
+        let t5 = compute_base4_conversion_table::<64>();
+        let t6 = compute_base4_conversion_table::<128>();
+        let t7 = compute_base4_conversion_table::<256>();
+        let t8 = compute_base4_conversion_table::<512>();
+
+        println(f"pub global BASE4_ENCODE_1BIT_TABLE: [Field; 2] = {t0};");
+        println(f"pub global BASE4_ENCODE_2BIT_TABLE: [Field; 4] = {t1};");
+        println(f"pub global BASE4_ENCODE_3BIT_TABLE: [Field; 8] = {t2};");
+        println(f"pub global BASE4_ENCODE_4BIT_TABLE: [Field; 16] = {t3};");
+        println(f"pub global BASE4_ENCODE_5BIT_TABLE: [Field; 32] = {t4};");
+        println(f"pub global BASE4_ENCODE_6BIT_TABLE: [Field; 64] = {t5};");
+        println(f"pub global BASE4_ENCODE_7BIT_TABLE: [Field; 128] = {t6};");
+        println(f"pub global BASE4_ENCODE_8BIT_TABLE: [Field; 256] = {t7};");
+        println(f"pub global BASE4_ENCODE_9BIT_TABLE: [Field; 512] = {t8};");
```

Comment:
Can you replace these with assertions that `BASE4_ENCODE_1BIT_TABLE == t0`, etc. so these tests constrain the values of the globals to be correct?

Reply:
done